test/e2e/specs/format-api.test.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import {
+	clickBlockAppender,
+	META_KEY,
+	newPost,
+	pressWithModifier,
+} from '../support/utils';
+import { activatePlugin, deactivatePlugin } from '../support/plugins';
+
+/**
+ * The modifier keys needed to invoke a 'select the next word' keyboard shortcut.
+ *
+ * @type {string}
+ */
+const SELECT_WORD_MODIFIER_KEYS = process.platform === 'darwin' ? [ 'Shift', 'Alt' ] : [ 'Shift', 'Control' ];
+
+describe( 'Using Format API', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	afterAll( async () => {
+		await deactivatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	beforeEach( async () => {
+		await newPost();
+	} );
+
+	it( 'Format toolbar is present in a paragraph block', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'First paragraph' );
+		await page.mouse.move( 200, 300, { steps: 10 } );
+		expect( await page.$( '.editor-format-toolbar' ) ).not.toBeNull();
+	} );
+
+	it( 'Clicking the control wraps the selected text properly with HTML code', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'First paragraph' );
+		await pressWithModifier( SELECT_WORD_MODIFIER_KEYS, 'ArrowLeft' );
+		await pressWithModifier( META_KEY, 'A' );
+		await page.mouse.move( 200, 300, { steps: 10 } );
+		await page.click( '[aria-label="Custom Link"]' );
+		const paragraphContent = await page.$eval( 'div[data-type="core/paragraph"] p', ( element ) => element.innerHTML );
+		expect( paragraphContent ).toEqual( '<a href="#test" class="my-plugin-link" data-mce-selected=\"inline-boundary\">First paragraph</a>' );
```

Comment:
Could we use a snapshot here?

```js
import { getEditedPostContent } from '../support/utils';

expect( await getEditedPostContent() ).toMatchSnapshot();
```

Reply:
@raquelmsmith, can you update as suggested and I will merge it afterwards, thanks :)

Reply:
Sure, I'll take a stab at it 😅

Reply:
Just added the snapshot 👍 But I don't have experience with it, and it seemed a bit too simple... so not sure I did it right. Let me know if there is something I need to change.

Reply:
Perfect. But now you can remove these two lines as they check the same thing?

Reply:
Yes, we can remove the following lines:

```js
const paragraphContent = await page.$eval( 'div[data-type="core/paragraph"] p', ( element ) => element.innerHTML );
expect( paragraphContent ).toEqual( '<a href="#test" class="my-plugin-link" data-mce-selected=\"inline-boundary\">First paragraph</a>' );
```